pettingzoo/magent/magent_env.py

```diff
@@ -127,6 +145,40 @@ def _all_dones(self, step_done=False):
         ret_agents = set(self.agents)
         return {agent: bool(done) for agent, done in zip(self.possible_agents, dones) if agent in ret_agents}
 
+    def state(self):
+        '''
+        Returns an observation of the global environment
+        '''
+        state = np.copy(self.base_state)
+
+        for handle in self._all_handles:
+            view, features = self.env.get_observation(handle)
+
+            pos = self.env.get_pos(handle)
+            pos_x, pos_y = zip(*pos)
+            state[pos_x, pos_y, 1 + handle.value * 2] = 1
+            state[pos_x, pos_y, 2 + handle.value * 2] = np.copy(view[:, view.shape[1] // 2, view.shape[2] // 2, 2])
+
+            if self.extra_features:
+                add_zeros = np.zeros((features.shape[0], state.shape[2] - (1 + len(self.team_sizes) * 2 + features.shape[1])))
```

Comment:
I'm confused why these add_zeros are necessary

Reply:
This is basically for the one_hot action feature. Some of the environments don't have the same action space for each team. I'm padding with zeros the one_hot_action of the agents with the lower action space size to match the other agents. add_zeros is the number of elements to be added

Reply:
Makes sense.